systems/olcf-frontier/system.py

```diff
@@ -0,0 +1,384 @@
+            with open(compiler_cfg_path, "w") as f:
+                f.write(self.gcc_compiler_cfg())
+            selections.append(compiler_cfg_path)
+        compiler_cfg_path = self.next_adhoc_cfg()
+        with open(compiler_cfg_path, "w") as f:
+            f.write(self.rocmcc_cfg())
+        selections.append(compiler_cfg_path)
+
+        return selections
+
+    def rocmcc_cfg(self):
+        return f"""\
+compilers:
+  - compiler:
+      spec: rocmcc@{self.rocm_version}
+      paths:
+        cc:  /opt/rocm-{self.rocm_version}/bin/amdclang
+        cxx: /opt/rocm-{self.rocm_version}/bin/amdclang++
+        f77: /opt/rocm-{self.rocm_version}/bin/amdflang
+        fc:  /opt/rocm-{self.rocm_version}/bin/amdflang
+      operating_system: sles15
+      target: any
+      environment:
+        set:
+          RFE_811452_DISABLE: '1'
+          GCC_X86_64: /usr
+      extra_rpaths:
+      - /opt/rocm-{self.rocm_version}/lib
+      modules:
+      - PrgEnv-amd/8.6.0
+      - craype/2.7.33
+      - cray-dsmml/0.3.0
+      - cray-mpich/8.1.31
+      - cray-libsci/24.11.0
+      - amd/6.2.4
+      - craype-x86-trento
+      - cray-pmi/6.1.15
+      - libfabric
+      - xpmem
+"""
+
+    def cce_compiler_cfg(self):
+        return f"""\
+compilers:
+  - compiler:
+      spec: cce@{self.compiler_version}
+      paths:
+        cc: /opt/cray/pe/cce/{self.compiler_version}/bin/craycc
+        cxx: /opt/cray/pe/cce/{self.compiler_version}/bin/crayCC
+        f77: /opt/cray/pe/cce/{self.compiler_version}/bin/crayftn
+        fc: /opt/cray/pe/cce/{self.compiler_version}/bin/crayftn
+      operating_system: sles15
+      target: any
+      modules: []
+      environment:
+        set:
+          RFE_811452_DISABLE: '1'
+          GCC_X86_64: /usr
+          CRAYLIBS_X86_64: /opt/cray/pe/cce/{self.compiler_version}/cce/x86_64/lib
+        prepend_path:
+          LD_LIBRARY_PATH: /usr/lib64/gcc/x86_64-suse-linux/13
+      extra_rpaths:
+      - /usr/lib64/gcc/x86_64-suse-linux/13
+      modules:
+      - PrgEnv-cray/8.6.0
+      - craype/2.7.33
+      - cray-dsmml/0.3.0
+      - cray-mpich/8.1.31
+      - cray-libsci/24.11.0
+      - cce/18.0.1
+      - craype-x86-trento
+      - cray-pmi/6.1.15
+      - libfabric
+      - xpmem
+"""
+
+    def gcc_compiler_cfg(self):
+        return f"""\
+compilers:
+  - compiler:
+      spec: gcc@{self.compiler_version}
+      paths:
+        cc: /usr/bin/gcc-{self.compiler_version}
+        cxx: /usr/bin/g++-{self.compiler_version}
+        f77: /usr/bin/gfortran-{self.compiler_version}
+        fc: /usr/bin/gfortran-{self.compiler_version}
+      operating_system: sles15
+      target: any
+      extra_rpaths: []
+      modules:
+      - PrgEnv-gnu/8.6.0
+      - craype/2.7.33
+      - cray-dsmml/0.3.0
+      - cray-mpich/8.1.31
+      - cray-libsci/24.11.0
+      - gcc-native/13.2
+      - craype-x86-trento
+      - cray-pmi/6.1.15
+      - libfabric
+      - xpmem
+"""
```

Comment:
This defs appear similar to compilers used by https://github.com/LLNL/benchpark/blob/develop/systems/llnl-elcapitan/system.py. The compiler specifications for benchpark changed in #953. These would need to be rewritten using the `compiler_def`/`compiler_section_for` functions introduced there (along with `merge_dicts` to get the final config. I think the elcap system definition linked here would be a good example to extrapolate from (if you want me to more explicitly detail what those changes would look like, let me know).